File: mns-reactor/src/test/java/cz/fb/manaus/reactor/categorizer/LayPriceCountCategorizerTest.java
package cz.fb.manaus.reactor.categorizer;

import cz.fb.manaus.core.category.BetCoverage;
import cz.fb.manaus.core.model.Side;
import cz.fb.manaus.core.test.AbstractLocalTestCase;
import cz.fb.manaus.core.test.CoreTestFactory;
import org.hamcrest.CoreMatchers;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.Set;

import static org.junit.Assert.assertThat;

public class LayPriceCountCategorizerTest extends AbstractLocalTestCase {

    @Autowired
    private LayPriceCountCategorizer categorizer;

    @Test
    public void testCategory() throws Exception {
        assertThat(categorizer.getCategories(CoreTestFactory.newSettledBet(2d, Side.LAY), BetCoverage.EMPTY),
                CoreMatchers.is(Set.of("priceCountLay_0")));
    }
}

File: mns-reactor/src/test/java/cz/fb/manaus/reactor/categorizer/LayPriceCountCategorizerTest.kt
package cz.fb.manaus.reactor.categorizer

import cz.fb.manaus.core.category.BetCoverage
import cz.fb.manaus.core.model.realizedBet
import cz.fb.manaus.core.test.AbstractLocalTestCase
import org.junit.Test
import org.springframework.beans.factory.annotation.Autowired
import kotlin.test.assertEquals

class LayPriceCountCategorizerTest : AbstractLocalTestCase() {

    @Autowired
    private lateinit var categorizer: LayPriceCountCategorizer

    @Test
    fun category() {
        assertEquals(setOf("priceCountLay_2"), categorizer.getCategories(realizedBet, BetCoverage.EMPTY))
    }
}